src/Symfony/Bundle/FrameworkBundle/Secrets/SodiumVault.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Secrets;
+
+/**
+ * @author Tobias Schultze <http://tobion.de>
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class SodiumVault
+{
+    private $encryptionKey;
+    private $decryptionKey;
+    private $pathPrefix;
+
+    /**
+     * @param string|object|null $decryptionKey A string or a stringable object that defines the private key to use to decrypt the vault
+     *                                          or null to store them in the provided $secretsDir
+     */
+    public function __construct(string $secretsDir, $decryptionKey = null)
```

Comment:
The `$decryptionKey` argument seems to be detached from the instance, aren't we missing some `$this->decryptionKey = $decryptionKey` here?